src/main/scala/org/apache/flinkx/api/serializer/CaseClassSerializer.scala

```diff
@@ -37,95 +37,69 @@ abstract class CaseClassSerializer[T <: Product](
 ) extends TupleSerializerBase[T](clazz, scalaFieldSerializers)
     with Cloneable {
 
-  @transient var fields: Array[AnyRef] = _
-
-  @transient var instanceCreationFailed: Boolean = false
-
   @transient lazy val log: Logger = LoggerFactory.getLogger(this.getClass)
 
+  override def isImmutableType(): Boolean = true
```

Comment:
```suggestion
  override def isImmutableType(): Boolean = scalaFieldSerializers.forall(_.isImmutableType)
```
I got an idea about cases I mentioned [here](https://github.com/flink-extended/flink-scala-api/pull/113#issuecomment-2095049462), it'd be better if some fields of a case class is mutable then this whole class becomes mutable

Reply:
ok. Done